### ISSUE TITLE
Cannot Debug on iPhone device

### ISSUE THREAD

**Issue by Amd3202** [xamarin/xamarin-macios#3915]
Building the app succeeds. App gets copied/installed to iPhone Device successfully. 

App FAILS to launch from VS and no debug info is captured. If you manually launch the app that was just copied it works fine, but again no debug info is sent to VS2017. 

1. Build in debug, application loads onto the device, however fails to launch.

BUILD: 
 PostflightingApplication - PercentComplete: 70%
 SandboxingApplication - PercentComplete: 80%
 GeneratingApplicationMap - PercentComplete: 90%
Application bundle 'com.eimc.mobile' installed on 'Ad iPhone2'

DEBUG:
The app has been terminated.
Launch failed. The app 'EIMC.iOS' could not be launched on 'Ad iPhone2'. Error: error MT1006: Could not install the application '/Users/amitlaptop/Library/Caches/Xamarin/mtbs/builds/EIMC.iOS/3f7616e5ab8de1149b09da6521379345/bin/iPhone/Debug/EIMC.iOS.app' on the device 'Ad iPhone2': Read Error (error: 0xe8000004).. Please check the logs for more details.
The app has been terminated.
The app has been terminated.

ERROR List (entire Solution):
Severity	Code	Description	Project	File	Line	Suppression State
Error		error MT1006: Could not install the application '/Users/amitlaptop/Library/Caches/Xamarin/mtbs/builds/EIMC.iOS/3f7616e5ab8de1149b09da6521379345/bin/iPhone/Debug/EIMC.iOS.app' on the device 'Ad iPhone2': Read Error (error: 0xe8000004).		 	0	

### Environment

```
```

**Comment by jsauve** [comment]
Same here. My deploy-to-device step would take forever, and then fail with a "read error". Must have been my USB-C to lighting cable. In my case, I just had to flip the lightning cable around. I eventually switched to a different cable altogether.

**Comment by Amd3202** [comment]
While i experience similar issues, i really don't think it's the cable. I can reboot without removing the cable and it's work fine. But yes, the copyingfile - PercentComplete takes forever and then fails. It's happening right now infact. I tried your trick of flipping the cable to see but not helping. The issue started right after i installed an ad-hoc release build from MS-AppCenter and then went back to local/Debug builds. my app icon has now gone to the default white with the circle-grid icon and the MT1006: could not install happens on ever build... switching to Android for now :)

**Comment by Amd3202** [comment]
Well now the application builds and installs in a few seconds, per usual, but there is no debug information coming back to VS... will pick up another new cable tomrorow, but using a fairly new one direct from apple not a generic brand...

**Comment by jsauve** [comment]
Same here. Mine was official Apple as well.

**Comment by Amd3202** [comment]
Thanks for your help and insight. I HOPE a $27 cable fixes my woes! Have lost a bunch of hours today on this. not to mention over the last few weeks! Thanks have a great night.

**Comment by Amd3202** [comment]
Last one... just tried via network (no cable) and same cannot write error...

**Comment by jsauve** [comment]
And that's with WiFi debugging enabled, I would assume?

I frequently do a "deep clean" by completely deleting all `bin` & `obj` folders (not via the Clean action in VS). I have a blog post about creating a right-click action in Finder to do this:
http://www.joesauve.com/remove-obj-and-bin-folders-via-os-x-finder/

Just be sure to set the constrained path in the script to the folder in which you keep all your source code projects.

**Comment by Amd3202** [comment]
Yep very Familiar with that and then reboot Or at least restart VS often... let’s see what happens with a new cable tomorrow thanks again

**Comment by jsauve** [comment]
Wow. Very interesting. The Xamarin tooling definitely breaks sometimes. You could try switching to beta or alpha update channel momentarily, just to see if those release pipelines possibly have a fix already.

**Comment by Amd3202** [comment]
app icon is still missing though...

**Comment by MaxxDelusional** [comment]
My app was deploying fine when I left work last night, but would not deploy to my device when I came in this this morning. I did not touch the phone, or making any code changes during this time.

Using @Amd3202 suggestion, I manually changed the Build number in Info.plist, and my app instantly started to deploy again.

**Comment by alwydl** [comment]
None of the approachs worked for me.  I tried:
- reseting the phone
- changing build number
- debugging through Wi-Fi

Exactly same error. MT1006 and 0xe8000004

Any other ideas?

**Comment by Amd3202** [comment]
I assume you have upgraded to latest versions of VS? There were definitely some builds there that VS was totally crashing with Xamarin. I also heard from a MS MVP that the last build has been problematic, so i have not upgraded... 2/18/2019.... also make sure you're in Debug not Release. EVERY other time i had this issue it was related to the version and build number. Make sure you're updating both of them since i really can't tell what jams it up. Best of luck and please let us know (below) if you find something else that solves it.

**Comment by rolfbjarne** [comment]
Does it work if you create a new app from a template?

**Comment by alwydl** [comment]
It actually started working again. But it made no sense.
It was not working on VS 2017 Version 15.9.3, Xam.iOS 12.2.1.15
I added the build version - Did Not Work
Went to cable - Did Not Work
Tried from Notebook B with VS 2017 Version 15.9.7 - Didi Not work

**Comment by alwydl** [comment]
One more piece of information. Changed link from SDK only to none, it failed again. Changed back to Link SDK's only, it got bat working. (Maybe this help someone)

**Comment by diogopio** [comment]
Try creating a new project in XCode and try to run on the same device you were trying to run with Xamarin, in my case, there was something about signing to be fixed. XCode fixed and after that, I was able to run the Xamarin app.
Hope it can help someone!

**Comment by amirvenus** [comment]
I am still having the same issue!

**Comment by marceloarchiza** [comment]
Same issue

**Comment by lhughey** [comment]
> My app was deploying fine when I left work last night, but would not deploy to my device when I came in this this morning. I did not touch the phone, or making any code changes during this time.
> 
> Using @Amd3202 suggestion, I manually changed the Build number in Info.plist, and my app instantly started to deploy again.

The solution of changing the build number currently works (using the latest Xamarin versions).

**Comment by wislon** [comment]
I ran into this problem this morning as well, after upgrading to the latest stable VS 2019 (15.5), 

I changed the linker settings to 'Do not link', and changed the build number in the `Info.plist` (I went from `1.0` to `1.0.1`), and it immediately worked after a clean and rebuild.
